ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/Network/NodeToNode.hs

```diff
@@ -408,6 +414,12 @@ defaultCodecs ccfg version encAddr decAddr nodeToNodeVersion =
           dec
           enc
           dec
+    , cPerasCertDiffusionCodec =
+        codecObjectDiffusion
+          encodePerasRoundNo
+          decodePerasRoundNo
+          (encodePerasCert p)
+          (decodePerasCert p)
```

Comment:
Please correct me if I'm wrong: having a `SerializeNodeToNode` instance for this (as @amesgen (suggested)[https://github.com/IntersectMBO/ouroboros-consensus/pull/1658/files#r2319079106)) would also allow us to consistently use the overloaded `enc` and `dec` helpers here, no?

Reply:
Exactly :+1:

For reference:
https://github.com/IntersectMBO/ouroboros-consensus/blob/59213abc6393ae9ed97b80a9b117ee82c4104ad7/ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/Network/NodeToNode.hs#L385-L389